test/export/test_funtionalized_assertions.py

```diff
@@ -26,6 +39,38 @@ def test_functional_sym_constrain_range(self) -> None:
             dep_token,
         )
 
+    def test_eager_mode_functionalization(self) -> None:
+        def my_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
+            FileCheck().check_count(
+                "aten._make_dep_token.default", 1, exactly=True,
+            ).run(gm.code)
+            FileCheck().check_count(
+                "aten._functional_assert_async.msg", 1, exactly=True,
+            ).run(gm.code)
+            return make_boxed_func(gm.forward)
+
+        my_compiler = aot_autograd(fw_compiler=my_compiler)
+
+        def f(x):
+            b = x.sin()
+            assert x[0] == 3
+            return x.cos() + b
+
+        with _enable_functionalization():
+            compiled = torch.compile(f, backend=my_compiler)
+            inp = torch.Tensor([3, 4, 5])
+            self.assertTrue(torch._dynamo.utils.same(compiled(inp), f(inp)))
```

Comment:
This is a great E2E test. If would be great if you could also a test more directly for AOTAutograd

(1) add a similar test in `test/functorch/test_aotdispatch.py`, that calls `aot_export_module()`
(2) include an expecttest in that test, that asserts on what the graph looks like (to show that is has a functional assert call in it). There are some examples in that file, e.g. [here](https://github.com/pytorch/pytorch/blob/1a8af1503f10540c4db229ea95a72a32340a965f/test/functorch/test_aotdispatch.py#L2027)